gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -967,25 +967,920 @@ public void testListDnsRecords() {
   }
 
   @Test
-  public void testListZoneBatch() {
-    DNS.create(ZONE1);
-    DNS.create(ZONE_EMPTY_DESCRIPTION);
+  public void testListZonesBatch() {
     try {
       DnsBatch batch = DNS.batch();
-      DnsBatchResult<Page<Zone>> batchResult = batch.listZones();
+      DnsBatchResult<Page<Zone>> result = batch.listZones();
       batch.submit();
-      assertTrue(batchResult.completed());
-      Iterator<Zone> iteratorBatch = batchResult.get().iterateAll();
-      Iterator<Zone> iteratorList = DNS.listZones().iterateAll();
-      assertEquals(Iterators.size(iteratorList), Iterators.size(iteratorBatch));
-      while (iteratorBatch.hasNext()) {
-        assertTrue(Iterators.contains(iteratorList, iteratorBatch.next()));
+      List<Zone> zones = filter(result.get().iterateAll());
+      assertEquals(0, zones.size());
+      // some zones exists
+      Zone created = DNS.create(ZONE1);
+      batch = DNS.batch();
+      result = batch.listZones();
+      batch.submit();
+      zones = filter(result.get().iterateAll());
+      assertEquals(created, zones.get(0));
+      assertEquals(1, zones.size());
+      created = DNS.create(ZONE_EMPTY_DESCRIPTION);
+      batch = DNS.batch();
+      result = batch.listZones();
+      DnsBatchResult<Page<Zone>> zeroSizeError =
+          batch.listZones(Dns.ZoneListOption.pageSize(0));
+      DnsBatchResult<Page<Zone>> negativeSizeError =
+          batch.listZones(Dns.ZoneListOption.pageSize(-1));
+      DnsBatchResult<Page<Zone>> okSize = batch.listZones(Dns.ZoneListOption.pageSize(1000));
```

Comment:
I would slightly prefer to see page size of 1, and then you can assert down below that you only get one result.